src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java

```diff
@@ -147,4 +157,34 @@ private void deduceDataClassificationForNewFields(CaseType caseType, CaseDetails
         jsonNode.ifPresent(value -> data.remove(CALLBACK_RESPONSE_KEY_STATE));
         return jsonNode.flatMap(value -> value.isTextual() ? Optional.of(value.textValue()) : Optional.empty());
     }
+
+    private void validateSignificantItem(CaseEvent eventTrigger, CallbackResponse callbackResponse) {
+        SignificantItem significantItem = callbackResponse.getSignificantItem();
+        List<String> errors = new ArrayList<>();
+
+        if (significantItem != null){
+            if(significantItem.getType() != ItemType.DOCUMENT){
+
+                errors.add("Significant Item type incorrect");
+            }
+            UrlValidator urlValidator = new UrlValidator();
```

Comment:
Is this threadsafe? If so it should be moved to a static variable.

Reply:
Still relevant

Reply:
Is it URLValidator you are referring to , if so its a third party apache utils class

Reply:
I have changed it to   
private static final UrlValidator URL_VALIDATOR = UrlValidator.getInstance();
if (!URL_VALIDATOR.isValid(significantItem.getUrl())) {, 
but it could be just 
if (!UrlValidator.getInstance().isValid(significantItem.getUrl())) {